x-pack/test/api_integration/apis/ml/trained_models/get_models.ts

```diff
@@ -0,0 +1,88 @@
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  describe('GET trained_models', () => {
+    let testModelIds: string[] = [];
+
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      testModelIds = await ml.api.createdTestTrainedModels('regression', 5, true);
+      await ml.api.createModelAlias('dfa_regression_model_n_0', 'dfa_regression_model_alias');
+      await ml.api.createIngestPipeline('dfa_regression_model_alias');
+    });
+
+    after(async () => {
+      // delete created ingest pipelines
+      await Promise.all(
+        ['dfa_regression_model_alias', ...testModelIds].map((modelId) =>
+          ml.api.deleteIngestPipeline(modelId)
+        )
+      );
+      await ml.api.cleanMlIndices();
+    });
+
+    it('returns all trained models with associated pipelines including aliases', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/trained_models?with_pipelines=true`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+      // Created models + system model
+      expect(body.length).to.eql(6);
+
+      const sampleModel = body.find((v: any) => v.model_id === 'dfa_regression_model_n_0');
+      expect(Object.keys(sampleModel.pipelines).length).to.eql(2);
+    });
+
+    it('returns models without pipeline in case user does not have required permission', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/trained_models?with_pipelines=true`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+      // Created models + system model
+      expect(body.length).to.eql(6);
+      const sampleModel = body.find((v: any) => v.model_id === 'dfa_regression_model_n_0');
+      expect(sampleModel.pipelines).to.eql(undefined);
+    });
+
+    it('returns trained model by id', async () => {
+      const { body } = await supertest
+        .get(`/api/ml/trained_models/dfa_regression_model_n_1`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+      expect(body.length).to.eql(1);
+      expect(body[0].model_id).to.eql('dfa_regression_model_n_1');
+    });
+
+    it('returns 404 if requested trained model does not exist', async () => {
+      await supertest
+        .get(`/api/ml/trained_models/not_existing_model`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(404);
+    });
+
+    it('return an error for unauthorized user', async () => {
```

Comment:
super duper nit: should this be `returns`?

Reply:
Fixed in https://github.com/elastic/kibana/pull/104819/commits/407294afab4bd42e29d5d806f928d14b296f5490